### PR TITLE
CLOUDP-78289: cr annotations update

### PR DESCRIPTION
**Pull request by FlorentinaSimlinger**
I was not able to reproduce the bug. I changed the spec.version field in the cr yaml file multiple times and it correctly updated the annotation fields lastSuccessfulConfiguration and lastAppliedMongoDBVersion.

However, in the process of debugging I noted the interchangeable use of "last" and "previous" as well as the occasional use of (in this case) ambiguous "old" and "new". 
This was confusing as the field lastSuccessfulConfiguration only becomes the "previous" config once the reconciliation is done/the configuration is successful. Before the reconciliation is done, it is just the current config.
I tried to clear this ambiguity by using "attempted" for the new spec we want, "last successful" for the spec that is in place until it is replaced, and generally avoided the use of "last" for "previous".

I spent quite some time tracing the changing meaning of lastSuccessfulConfiguration and lastAppliedMongoDBVersion and its use in the code, so what I'm hoping for is that it's easier to follow along now. Let me know what you think.

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you signed all of your commits?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: controllers/replica_set_controller.go [hunk 1]
```diff
@@ -248,7 +248,8 @@ func (r ReplicaSetReconciler) Reconcile(ctx context.Context, request reconcile.R
 	return res, err
 }
 
-// updateLastSuccessfulConfiguration annotates the MongoDBCommunity resource with the latest configuration
+// updateLastSuccessfulConfiguration annotates the MongoDBCommunity resource with the configuration
+// that was used to create the resource
```
Comment (irajdeep) [inline controllers/replica_set_controller.go:252]: > that was used to create the resource

This is called for every reconcile(whenever the spec is updated) -- why is it only annotating with the configuration that was used to create the resource?

Comment (FlorentinaSimlinger) [reply controllers/replica_set_controller.go:252]: Good point

File: api/v1/mongodbcommunity_types.go [hunk 2]
```diff
@@ -592,12 +592,13 @@ func (m MongoDBCommunity) GetUpdateStrategyType() appsv1.StatefulSetUpdateStrate
 
 // IsChangingVersion returns true if an attempted version change is occurring.
 func (m MongoDBCommunity) IsChangingVersion() bool {
-	prevVersion := m.getPreviousVersion()
-	return prevVersion != "" && prevVersion != m.Spec.Version
+	lastVersion := m.getLastVersion()
+
```
Comment (irajdeep) [inline api/v1/mongodbcommunity_types.go:596]: Let's remove this empty line

File: controllers/replica_set_controller.go [hunk 3]
```diff
@@ -261,6 +261,7 @@ func (r *ReplicaSetReconciler) updateLastSuccessfulConfiguration(mdb mdbv1.Mongo
 		// This is needed to reuse the update strategy logic in enterprise
 		lastAppliedMongoDBVersion: mdb.Spec.Version,
 	}
+
```
Comment (irajdeep) [inline controllers/replica_set_controller.go:264]: same

**Review by irajdeep — APPROVED** [review]
LGTM

File: controllers/validation/validation.go [hunk 4]
```diff
@@ -9,12 +9,12 @@ import (
...
 )
 
-// ValidateInitalSpec checks if the resource initial Spec is valid
+// ValidateInitalSpec checks if the resource's initial Spec is valid.
 func ValidateInitalSpec(mdb mdbv1.MongoDBCommunity) error {
 	return validateSpec(mdb)
 }
 
-// ValidateUpdate validates that the new Spec, corresponding to the existing one is still valid
+// ValidateUpdate checks if the attempted Spec is valid and if it is invalidated by the last successful Spec's Security field.
```
Comment (irajdeep) [inline controllers/validation/validation.go:17]: let's revert this comment -- I think the old one is more apt

**Pull request merged**